src/app/static/src/app/store/modelOutput/modelOutput.ts

```diff
@@ -5,7 +5,9 @@ import {FilterOption} from "../../generated";
 
 const namespaced: boolean = true;
 
-export interface ModelOutputState {}
+export interface ModelOutputState {
+    dummyProperty: boolean // unused except for testing
```

Comment:
Why do we have an empty module here, and should the bar chart plotting selections stuff rightly live in this module?

Reply:
Hmm.. I think you could argue that the selections are different to the model output since they're modified by the user rather than coming from the model output. But maybe it would make sense for the modelRun.result to inhabit modelOutput rather than modelRun - I've always found that a bit confusing, though I can see why it happened.

Reply:
> One thing I noticed is that if I go back to the options after being on a subsequent step and edit them, the output steps become invalidated as expected, but I don't need to re-validate the options before continuing. Probably should have to though?

Thanks for picking this up, yes this was an oversight. But we're actually going to ditch the validation altogether though I think worth just leaving on this branch as is. Ticket for removing the validation logic - https://vimc.myjetbrains.com/youtrack/issue/mrc-792

> Is there a separate ticket for the issue where the choropleth disappears when you remove survey, but have other surveyandProgram files present? 

Ha, no, this was supposed to be working here! Clearly my tests are not doing their job :woman_facepalming: 

> Oh, and one other little thing I've just noticed, not directly related to this ticket, is that if you have an error with anc or programme, you can still continue to next step, because survey is valid. Maybe this is ok because I can't actually 'remove' the errored file, and maybe I don't actually have a valid one and just hit the wrong upload button. It just seems a little odd to be able to continue with errors. What do you think?

Good question, does seem like we should block continuing if in error state. Added a ticket: https://vimc.myjetbrains.com/youtrack/issue/mrc-791

Reply:
I was thinking of the modules as roughly corresponding to steps of the app, not sure there's a reason to distinguish user mutated from api mutated content. Anyway, just something to consider refactoring when we get a chance! I agree, result should also be in modelOutput, it's confusing as is!